airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java

```diff
@@ -357,7 +357,33 @@ public ManualSyncSubmissionResult resetConnection(final UUID connectionId) {
       }
     } while (connectionManagerWorkflow.getJobInformation().getJobId() == oldJobId);
 
-    log.info("end of reset");
+    log.info("end of reset submission");
+
+    final long jobId = connectionManagerWorkflow.getJobInformation().getJobId();
+
+    return new ManualSyncSubmissionResult(
+        Optional.empty(),
+        Optional.of(jobId));
+  }
+
+  public ManualSyncSubmissionResult synchronousResetConnection(UUID connectionId) {
```

Comment:
I think this needs a doc string. It isn't intuitive from reading that the condition it's waiting for is that the connection manager switches to a new job after completing the reset flow.

Reply:
Done